### ISSUE TITLE
Controller crashes on pods with Linkerd sidecar proxy containers

### ISSUE THREAD

**Issue by shandr** [enix/kube-image-keeper#419]
Hi! Controllers crash with the following error when working with pods with Linkerd side car enabled.

```
2024-10-03T12:56:43.683Z    ERROR    setup    problem running manager    {"error": "Pod \"wiremock-s2-7bf6c79684-geiwr\" is invalid: [spec.initContainers[1].livenessProbe: Forbidden: may not be set for init containers without restartPolicy=Always, spec.initContainers[1].readinessProbe: Forbidden: may not be set for init containers without restartPolicy=Always, spec.initContainers[1].startupProbe: Forbidden: may not be set for init containers without restartPolicy=Always, spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`,`spec.initContainers[*].image`,`spec.activeDeadlineSeconds`,`spec.tolerations` (only additions to existing tolerations),`spec.terminationGracePeriodSeconds`
```
Kubernetes: 1.29
kube-image-keeper:1.9.3

**Comment by kppullin** [comment]
This seems like the issue from https://github.com/enix/kube-image-keeper/issues/371 , with a workaround in https://github.com/enix/kube-image-keeper/pull/397 .

The workaround isn't in 1.9.3... so you can try building a custom image from `main` or wait for the next release (hopefully they'll cut a new release soon 😄 ).

Alternatively, have the linkerd webhook ignore pod patch events... but I don't know if this is possible nor risks it may raise today or in the future.

**Comment by Nicolasgouze** [comment]
hi @shandr , @kppullin, we released last week the 1.10 release, including your PR @kppullin .

**Comment by kppullin** [comment]
Thanks - it does appear to have solved the issue.
